utils/s3/credentials_providers/aws_credentials_provider_chain.hh

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+#include "aws_credentials_provider.hh"
+
+namespace aws {
+
+class aws_credentials_provider_chain final {
+public:
+    [[nodiscard]] seastar::future<s3::aws_credentials> get_aws_credentials();
```

Comment:
`class future<>` itself is `[[nodiscard]]`, why one more here?

Reply:
I guess I was returning just `s3::aws_credentials` in the first place. removing